src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -1373,6 +1376,582 @@ private Autogenerated.StateItem ToAutogeneratedStateItem(StateTransactionRequest
+                    yield return Array.Empty<byte>();
+                }
+            }
+
+            //Send the completion message
+            await duplexStream.RequestStream.CompleteAsync();
+            yield break; //End the async enumeration
+        }
+
+        /// <summary>
+        /// Retrieves the encrypted bytes from the encryption operation on the sidecar and returns as an enumerable stream.
+        /// </summary>
+        private async IAsyncEnumerable<byte[]> RetrieveEncryptedStreamAsync(AsyncDuplexStreamingCall<Autogenerated.EncryptRequest, Autogenerated.EncryptResponse> duplexStream, [EnumeratorCancellation] CancellationToken cancellationToken)
+        {
+            await foreach (var encryptResponse in duplexStream.ResponseStream.ReadAllAsync(cancellationToken)
+                               .ConfigureAwait(false))
+            {
+                yield return encryptResponse.Payload.Data.ToByteArray();
+            }
+        }
+        
+        /// <inheritdoc />
+        [Obsolete(
+            "The API is currently not stable as it is in the Alpha stage. This attribute will be removed once it is stable.")]
+        public override async IAsyncEnumerable<ReadOnlyMemory<byte>> DecryptAsync(string vaultResourceName, Stream ciphertextStream, string keyName,
+            DecryptionOptions decryptionOptions, [EnumeratorCancellation] CancellationToken cancellationToken = default)
+        {
+            ArgumentVerifier.ThrowIfNullOrEmpty(vaultResourceName, nameof(vaultResourceName));
+            ArgumentVerifier.ThrowIfNullOrEmpty(keyName, nameof(keyName));
+            ArgumentVerifier.ThrowIfNull(ciphertextStream, nameof(ciphertextStream));
+            ArgumentVerifier.ThrowIfNull(decryptionOptions, nameof(decryptionOptions));
+
+            var decryptRequestOptions = new Autogenerated.DecryptRequestOptions
+            {
+                ComponentName = vaultResourceName, KeyName = keyName
+            };
+
+            var options = CreateCallOptions(headers: null, cancellationToken);
+            var duplexStream = client.DecryptAlpha1(options);
+
+            //Run both operations at the same time, but return the output of the streaming values coming from the operation
+            var receiveResult = Task.FromResult(RetrieveDecryptedStreamAsync(duplexStream, cancellationToken));
+            await Task.WhenAll(
+                //Stream the ciphertext data to the sidecar in chunks
+                Task.FromResult(SendCiphertextStreamAsync(ciphertextStream, decryptionOptions.StreamingBlockSizeInBytes,
+                    duplexStream, decryptRequestOptions, cancellationToken)),
+                //At the same time, retrieve the decrypted response from the sidecar
+                receiveResult
+            );
+            
+            //Return only the result of the `RetrieveEncryptedStreamAsync` method
+            await foreach (var item in receiveResult.Result.WithCancellation(cancellationToken))
```

Comment:
It occurs to me that, if we're re-iterating and re-yielding the `receiveResult`, do we need to wait for either task above?  Alternatively, can `receiveResult.Result` just be returned without the iteration?

Reply:
Updated accordingly - removed the second (and unnecessary) `await foreach` and updated the example.